apps/oracledb.go

```diff
@@ -0,0 +1,985 @@
+				monotonic:         "true",
+				value_type:        "int",
+				attribute_columns: []string{"DATABASE_ID", "GLOBAL_NAME", "INSTANCE_ID"},
+				static_attributes: map[string]string{
+					"db.system": "oracle",
+				},
+			},
+			{
+				metric_name:       "oracle.user.rollbacks",
+				value_column:      "USER_ROLLBACKS",
+				unit:              "{rollbacks}",
+				description:       "The total number of times users manually issue the ROLLBACK statement or an error occurs during a user's transactions",
+				data_type:         "sum",
+				monotonic:         "true",
+				value_type:        "int",
+				attribute_columns: []string{"DATABASE_ID", "GLOBAL_NAME", "INSTANCE_ID"},
+				static_attributes: map[string]string{
+					"db.system": "oracle",
+				},
+			},
+		},
+	},
+}
+
+func init() {
+	confgenerator.MetricsReceiverTypes.RegisterType(func() confgenerator.Component { return &MetricsReceiverOracleDB{} })
+}
+
+type LoggingProcessorOracleDBAlert struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+}
+
+func (lr LoggingProcessorOracleDBAlert) Type() string {
+	return "oracledb_alert"
+}
+
+func (lr LoggingProcessorOracleDBAlert) Components(tag string, uid string) []fluentbit.Component {
+	components := confgenerator.LoggingProcessorParseMultilineRegex{
+		LoggingProcessorParseRegexComplex: confgenerator.LoggingProcessorParseRegexComplex{
+			Parsers: []confgenerator.RegexParser{
+				{
+					// Sample log:  2021-12-21T10:19:47.339827-05:00
+					//				Thread 1 opened at log sequence 1
+					//				Current log# 1 seq# 1 mem# 0: /u01/oracle/oradata/DB19C/redo01.log
+					//				Successful open of redo thread 1
+					Regex: `^(?<timestamp>\d+-\d+-\d+T\d+:\d+:\d+.\d+(?:[-+]\d+:\d+|Z))\n(?<message>[\s\S]+)`,
+					Parser: confgenerator.ParserShared{
+						TimeKey:    "timestamp",
+						TimeFormat: "%Y-%m-%dT%H:%M:%S.%L%Z",
+					},
+				},
+			},
+		},
+		Rules: []confgenerator.MultilineRule{
+			{
+				StateName: "start_state",
+				NextState: "cont",
+				Regex:     `^\d+-\d+-\d+T\d+:\d+:\d+.\d+(?:[-+]\d+:\d+|Z)`,
+			},
+			{
+				StateName: "cont",
+				NextState: "cont",
+				Regex:     `^(?!\d+-\d+-\d+T\d+:\d+:\d+.\d+(?:[-+]\d+:\d+|Z)).*$`,
+			},
+		},
+	}.Components(tag, uid)
+
+	severityVal := "ALERT"
+	components = append(components,
+		confgenerator.LoggingProcessorModifyFields{
+			Fields: map[string]*confgenerator.ModifyField{
+				"severity":                 {StaticValue: &severityVal},
+				InstrumentationSourceLabel: instrumentationSourceValue(lr.Type()),
+			},
+		}.Components(tag, uid)...)
+	return components
+}
+
+type LoggingReceiverOracleDBAlert struct {
+	LoggingProcessorOracleDBAlert           `yaml:",inline"`
+	confgenerator.LoggingReceiverFilesMixin `yaml:",inline" validate:"structonly"`
+}
+
+func (lr LoggingReceiverOracleDBAlert) Components(tag string) []fluentbit.Component {
+	if len(lr.IncludePaths) == 0 {
+		lr.IncludePaths = []string{
+			"$ORACLE_HOME/diag/rdbms/*/*/trace/alert_*.log",
```

Comment:
If I'm assuming correctly that we're letting fluent-bit take care of filling the variable in, then this should be `${ORACLE_HOME}` (as well as the one on line 971).

Reply:
Good catch, fixing.